internal/services/account_member/resource.go

```diff
@@ -262,27 +305,36 @@ func (r *AccountMemberResource) ModifyPlan(ctx context.Context, req resource.Mod
 		return
 	}
 
-	// If destroying or creating, no need to modify plan
-	if config == nil || state == nil || plan == nil {
+	if config == nil && plan == nil {
+		// no plan changes for destroy
 		return
 	}
 
-	// Determine what's configured
-	configUsesRoles := config.Roles != nil && len(*config.Roles) > 0
-	configUsesPolicies := !config.Policies.IsNull() && !config.Policies.IsUnknown()
-
-	if configUsesRoles && !configUsesPolicies {
-		// When roles are configured, suppress policies diffs by keeping state value
-		plan.Policies = state.Policies
-	} else if configUsesPolicies && !configUsesRoles {
-		// When policies are configured, suppress roles diffs by keeping state value
-		plan.Roles = state.Roles
+	if state != nil {
+		// Always suppress user field diffs since it's computed and can change independently
+		// The user field contains computed values that may be updated by the API
+		resp.Diagnostics.Append(resp.Plan.SetAttribute(ctx, path.Root("user"), state.User)...)
 	}
 
-	// Always suppress user field diffs since it's computed and can change independently
-	// The user field contains computed values that may be updated by the API
-	plan.User = state.User
-
-	// Set the modified plan
-	resp.Diagnostics.Append(resp.Plan.Set(ctx, &plan)...)
+	if config != nil {
+		configurePermissionType := checkConfiguredPermissionType(config)
+		switch configurePermissionType {
+		case Roles:
+			if state == nil || !reflect.DeepEqual(plan.Roles, state.Roles) {
+				// if roles are changing, set policies to unknown
+				resp.Diagnostics.Append(resp.Plan.SetAttribute(ctx, path.Root("policies"), plan.Policies.UnknownValue(ctx))...)
```

Comment:
TBD: technically these "Unknown" cases should always be set. Every time we write the member back to the server, the server tries to rebuild roles/policies. Currently they won't ever change, but if we ever did change the server-side code that does that, then the values that come back will change which will cause terraform to crash (actual values != expected)